.circleci/config.yml

```diff
@@ -1269,6 +1269,46 @@ jobs:
           name: Verify Values Match
           command: |
             ./ops/scripts/ci-match-values-between-files.sh "<< parameters.file1_path >>" "<< parameters.pattern_file1 >>" "<< parameters.file2_path >>" "<< parameters.pattern_file2 >>"
+
+  test-proofs:
+    machine: true
+    steps:
+      - checkout
+      - check-changed:
+          patterns: contracts-bedrock/test/kontrol,contracts-bedrock/src/L1/OptimismPortal\.sol
+      - run:
+          name: Checkout submodules
+          command: |
+            git submodule sync
+            git submodule update --init --recursive
+      - run:
+          name: "Start Docker Container"
+          command: |
+            export KONTROL_VERSION=$(cat deps/kontrol_release)
+            docker run --name optimism-ci \
+                    --rm \
+                    -v $(pwd)/kout/proofs:/home/user/workspace/packages/contracts-bedrock/kout/proofs/ \
+                    --interactive \
+                    --tty \
+                    --detach \
+                    --user root \
+                    --workdir /home/user/workspace \
+                    runtimeverificationinc/kontrol:ubuntu-jammy-${KONTROL_VERSION}
+            docker cp . optimism-ci:/home/user/workspace
+            docker exec optimism-ci chown -R user:user /home/user
+      - run:
+          name: 'Run Kontrol'
+          command: |
+            docker exec -u user optimism-ci bash -c 'cd packages/contracts-bedrock && ./test/kontrol/kontrol/run-kontrol.sh'
+      - run:
+          name: Create Results Tar
+          command: |
+            tar -czvf results.tar.gz kout/proofs
+      - store_artifacts:
+          path: results.tar.gz
+          destination: results
+          when: always
```

Comment:
The proofs currently can take a few hours to run, so we upload the proof artifacts so CI, this way if a run fails we can download the artifacts and start debugging immediately instead of re-running the proofs locally.

The artifact files would be about 50-100MB in size, and we have CircleCI set to persist artifacts for 30 days. I want to confirm if this configuration is ok? @tynes @mslipper 

Once we get proof times down, we can consider removing this (but even still they'll likely take 20-30mins so it might be nice to keep it as-is)

Reply:
It would be nice to have a cost estimation of this. I looked and didn't seem to find a pricing model for artifacts.

Reply:
From https://app.circleci.com/settings/plan/github/ethereum-optimism/overview it says our plan gives us 200 GB of storage. And https://app.circleci.com/settings/plan/github/ethereum-optimism/usage says we're currently at 46.3 GB (denominated in "storage-months" which I'm guessing is related to the fact that by default artifacts last for 30 days). So as long as we stay under the 200GB limit I don't think it should add any cost

Reply:
It will only run on changes to affected contracts, which eventually will be the OptimismPortal, L1StandardBridge, L1ERC721Bridge, and the L1CrossDomainMessenger. For now we just have a dummy counter contract, then we'll incrementally add each of the above contracts. I think a good path forward here is:
- Switch to xlarge resource class to unblock this PR. These proofs will really only run once since it's a dummy contract for now.
- Monitor CI times as we start adding real proofs. If they become too long and/or too frequent, temporarily switch to a nightly CI job that creates an issue on failure
- Once kontrol has proof parallelization we can add proofs back into the standard PR flow and consider options for keeping credit usage low (such as offloading compute)